.devcontainer/Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Use the official OpenJDK image as base
+FROM mcr.microsoft.com/devcontainers/java:1-21-bookworm
+
+# Install additional tools
+RUN apt-get update && apt-get install -y \
+    git \
+    curl \
+    wget \
+    unzip \
+    && rm -rf /var/lib/apt/lists/*
+
+# Set up Node.js (needed for vaadin tests)
+RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash - \
+    && apt-get install -y nodejs
+
+# Install pnpm (needed for vaadin tests)
+RUN npm install -g pnpm
+
+# Set up gradle cache directory
+RUN mkdir -p /home/vscode/.gradle
+RUN chown -R vscode:vscode /home/vscode/.gradle
+
+# Switch to vscode user
+USER vscode
+
+# Set JAVA_HOME for vscode user
+ENV JAVA_HOME=/usr/lib/jvm/msopenjdk-current
+
+# Pre-configure git (optional, helps with development)
+RUN git config --global init.defaultBranch main
```

Comment:
this seems unnecessary in the context of a devcontainer which will be attached to a git repo already?